code/datums/components/material_container.dm

```diff
@@ -228,24 +302,25 @@
 	return total_amount_save - total_amount
 
 /// For spawning mineral sheets at a specific location. Used by machines to output sheets.
-/datum/component/material_container/proc/retrieve_sheets(sheet_amt, datum/material/M, target = null)
+/datum/component/material_container/proc/retrieve_sheets(sheet_amt, datum/material/M, atom/target = null)
 	if(!M.sheet_type)
 		return 0 //Add greyscale sheet handling here later
 	if(sheet_amt <= 0)
 		return 0
 
 	if(!target)
-		target = get_turf(parent)
+		var/atom/parent_atom = parent
+		target = parent_atom.drop_location()
```

Comment:
can we be sure that parent will always be an atom?

Reply:
Survey of the preexisting code says yes.

Reply:
if you're only going to support atoms then make it check atom in Initialize.

Reply:
Done